cachix-api/src/Cachix/Types/BinaryCache.hs

```diff
@@ -2,15 +2,32 @@ module Cachix.Types.BinaryCache where
 
 import Cachix.Types.Permission (Permission)
 import Data.Aeson (FromJSON, ToJSON)
-import Data.Swagger (ToSchema)
+import Data.Swagger (ToParamSchema (..), ToSchema)
 import Protolude
+import Servant.API
 
 data BinaryCache = BinaryCache
   { name :: Text,
     uri :: Text,
     isPublic :: Bool,
     publicSigningKeys :: [Text],
     githubUsername :: Text,
-    permission :: Permission
+    permission :: Permission,
+    compression :: CompressionMode
```

Comment:
Why does the binary cache report this? What is `compressionModeBackend` for?

Reply:
There's a binary cache setting to pick what compression should the client use. That can be overriden with `--compression-mode` on the CLI.

Reply:
Okay, so it's like a preferred encoding? But what's the intended use-case?

The way I see it, this remotely changes the client's default behavior, without actually enforcing that behavior. I worry that this may end up confusing users. Looking at this setting, I might ask:
* are all clients now forced to use this compression encoding?
* will all the NARs be stored in this encoding? Will they be re-encoded?
* will all the NARs be served in this encoding? 

Also,

1. How do we update the compression modes? Will we break older clients?
2. Should we instead let users permanently set the compression settings in their local config? They could either choose to use our defaults (ie, no preference), which we can change with new client releases, or they could request a specific mode (with levels, etc).